lib/hammer_cli/output/adapter/table.rb

```diff
@@ -68,73 +58,68 @@ def print_collection(all_fields, collection)
 
     protected
 
-    def field_filter
-      filtered = [Fields::ContainerField]
-      filtered << Fields::Id unless @context[:show_ids]
-      HammerCLI::Output::FieldFilter.new(filtered)
+    def normalize_column(width, value)
+      value = value.to_s
+      padding = width - HammerCLI::Output::Utils.real_length(value)
+      if padding >= 0
+        value += (" " * padding)
+      else
+        value, real_length = HammerCLI::Output::Utils.real_truncate(value, width-3)
+        value += '...'
+        value += ' ' if real_length < (width - 3)
+      end
+      value
     end
 
-    private
+    def format_values(fields, collection)
+      collection.collect do |d|
+        row = {}
+        fields.each do |f|
+          formatter = WrapperFormatter.new(@formatters.formatter_for_type(f.class), f.parameters)
+          row[f.label] = formatter.format(data_for_field(f, d) || "").to_s
+        end
+        row
+      end
+    end
 
-    def label_for(field)
-      width = width_for(field)
-      if width
-        "%-#{width}s" % field.label.to_s
-      else
-        field.label.to_s
+    def calculate_widths(fields, collection)
+      widths = {}
+      fields.map do |f|
```

Comment:
I think this can be onliner, but feel free to ignore if you find it unreadable
```ruby
Hash[fields.map { |f| [f.label, calculate_column_width(f, collection) }]
```